alphacsc/learn_d_z_multi.py

```diff
@@ -295,20 +283,18 @@ def _batch_learn(z_encoder, d_solver, end_iter_func, n_iter=100,
             print('[{}] CD iterations {} / {}'.format(name, ii, n_iter))
 
         if greedy and ii % n_iter_by_atom == 0 and \
-                z_encoder.D_hat.shape[0] < n_atoms:
+                d_solver.D_hat.shape[0] < n_atoms:
             # add a new atom every n_iter_by_atom iterations
-            new_atom = d_solver.get_max_error_dict(z_encoder)[0]
-            # XXX what should happen here when using DiCoDiLe?
-            z_encoder.add_one_atom(new_atom)
+            d_solver.add_one_atom(z_encoder)
 
-        if lmbd_max not in ['fixed', 'scaled']:
-            reg_ = reg * get_lambda_max(X, z_encoder.D_hat)
-            if lmbd_max == 'shared':
-                reg_ = reg_.max()
-            z_encoder.set_reg(reg_)
+        if lmbd_max in ['per_atom', 'shared'] or (
+                lmbd_max == 'scaled' and ii == 0
+        ):
+            z_encoder.update_reg()
```

Comment:
Here, `per_atom` is not respected anymore no?
I propose to either remove it or to pass `lmbda_max` as an argument?

Reply:
I did not undestand. 

There was `if lmbd_max=='shared'` line, do you mean that?

Reply:
yes

Reply:
That's fine like this, thx!